app/scripts/lib/rpc-method-middleware/handlers/log-web3-usage.js

```diff
@@ -45,10 +45,11 @@ function logWeb3UsageHandler (
   }
   if (!recordedWeb3Usage[origin][name]) {
     recordedWeb3Usage[origin][name] = true
+
     sendMetrics({
-      action,
-      name,
-      customVariables: { origin },
+      event: `Website Accessed window.web3`,
+      category: 'inpage_provider',
+      properties: { origin, action, web3Property: name },
```

Comment:
It would be ideal to set the `origin` as the referrer in the segment context, so that we can more easily group these events in Segment/Mixpanel.

The referrer URL is set on the `referrer.url` property of the context.

Reply:
2674ec5